build.assets/macos/tshdev/README.md

```diff
@@ -79,3 +80,116 @@ through the steps here unless you are trying to recreate parts of the skeleton
 
     For example, click "build" and copy the app using "Product -> Show Build
     Folder in Finder".
+
+## Working with launch daemons
+
+tsh.app includes a launch daemon for VNet under `Contents/Library/LaunchDaemons`. tsh uses
+[`SMAppService`](https://developer.apple.com/documentation/servicemanagement/smappservice?language=objc)
+to register the daemon plist. Once the daemon is registered and is enabled under login items, it can
+be started by sending a message to the XPC service advertised under `MachServices` in plist. This is
+going to make launchd start the program listed under `BundleProgram`. Extra arguments are passed to
+the program are under `ProgramArguments`. See `man launchd.plist` for more details behind individual
+plist properties.
+
+When there's no login item for a daemon and an app attempts to register it for the first time, macOS
+shows a notification about it which says "tsh.app added items that can run in the background for all
+users. Do you want to allow this?". When working on a launch daemon, at some point macOS is going to
+stop showing the notification. In Console.app, you're going to see this:
+
+```
+Exceeded max notifications for tsh. Please file a Feedback report or contact the developer.
+item=uuid=4DF802A6-A9DE-4EF8-A6FB-4B046E50E1DE, name=tsh, type=daemon, disposition=[enabled,
+disallowed, visible, not notified], identifier=com.goteleport.tshdev.vnetd,
+url=Contents/Library/LaunchDaemons/com.goteleport.tshdev.vnetd.plist -- file:///
+```
```

Comment:
what do you have to do at this point? could users hit this if they ignore the notification enough times? can we warn the user that this notification is going to come up and they need to click allow?

Reply:
If a notification doesn't get displayed, you pretty much just have to go to the login items yourself.

I think users are unlikely to hit this. Someone would have to move Connect/tsh to the trash, then empty the trash, go to Login Items, then install Connect/tsh again and start VNet and do it multiple times during the day. If you just trash Connect/tsh and then install it again without opening Login Items, then the background item stays enabled, I believe.

In any case, it's quite easy to miss the notification even when it's displayed, especially when using multiple monitors. I did it a couple of times when working on this PR. To help with this, I already added a message that's displayed by `tsh vnet` when we see that the daemon is waiting for an approval:

> To start VNet, please enable the background item for tsh.app in the Login Items section of System Settings.
> Waiting for the background item to be enabled…

Later I'm going to add an in-app notification to Connect which shows a similar message. This is especially important for Connect since the system notification looks like this:

<img width="367" alt="background-item" src="https://github.com/gravitational/teleport/assets/27113/cb718d3d-b2f0-40eb-97b5-9d48a364c5bd">

I think it's important for Connect to explicitly mention tsh.app, otherwise users could get suspicious and not understand why they got a notification for something called tsh.app while using Connect.